<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Diffusion Map on Single-cell Time Series Data Analysis</title>
    <meta charset="utf-8" />
    <meta name="author" content="Xinyi Lin" />
    <link href="libs/remark-css-0.0.1/default.css" rel="stylesheet" />
    <link href="libs/remark-css-0.0.1/metropolis.css" rel="stylesheet" />
    <link href="libs/remark-css-0.0.1/chocolate-fonts.css" rel="stylesheet" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# Diffusion Map on Single-cell Time Series Data Analysis
### Xinyi Lin
### Supervisor: Joshua Ho
### 2021/05/22 (updated: 2021-05-24)

---


&lt;style&gt;
p.caption {
  font-size: 0.6em;
}
&lt;/style&gt;

# Time Series Data

* data collected at different points in time(day1, day2, day3, day7......)

* some characteristics

  + potential for correlation between observations
  
  + regularly repeating pattern
  
&lt;img src="./image/JC0522/time_series.png" width="85%" style="display: block; margin: auto;" /&gt;

---

# Time Series Data

* data collected at different points in time(day1, day2, day3, day7......)

* some characteristics

  + potential for correlation between observations
  
  + regularly repeating pattern
  
&lt;img src="./image/JC0522/time_series2.png" width="100%" style="display: block; margin: auto;" /&gt;

---

# Diffusion Map

* dimensionality reduction/feature extraction algorithm

---

# Diffusion Map: Dimensionality Reduction

&lt;img src="./image/JC0522/difussion1.png" width="100%" style="display: block; margin: auto;" /&gt;

&lt;br/&gt;

.pull-left[
&lt;img src="./image/JC0522/difussion2.png" width="100%" style="display: block; margin: auto;" /&gt;
]

.pull-right[

* raw data are sparse

* save computational resources

* visualization
]

&lt;br/&gt;

&lt;p style="font-family: times, serif; font-size:11pt; font-style:italic"&gt;
Engel, D., Hüttenberger, L., &amp; Hamann, B. (2012). A survey of dimension reduction methods for high-dimensional data analysis and visualization.
&lt;/p&gt;

---

# Diffusion Map

* dimensionality reduction/feature extraction algorithm

* coordinates can be computed from the eigenvectors and eigenvalues of a diffusion operator on the data(non-linear version of PCA)

---

# Diffusion Map: non-linear version of PCA

PCA: creating new uncorrelated variables that successively maximize variance

.pull-left[
&lt;img src="./image/JC0522/difussion3.png" width="100%" style="display: block; margin: auto;" /&gt;
]

.pull-right[

* picked first few eigenvectors and eigenvalues

* computed coordinates from them

* projected to those coordinates
]

&lt;br/&gt;

-&gt; robust to noise perturbation

---

# Diffusion Map

* dimensionality reduction/feature extraction algorithm

* coordinates can be computed from the eigenvectors and eigenvalues of a diffusion operator on the data(non-linear version of PCA)

* the Euclidean distance between points in the embedded space is equal to the "diffusion distance" between probability distributions centered at those points


---

# Diffusion Map: Diffusion Framework

euclidean distance &lt;-&gt; difussion distance

.pull-left[
&lt;img src="./image/JC0522/difussion4.svg" width="100%" style="display: block; margin: auto;" /&gt;
]

.pull-right[
* running the chain forward in time geometric structure at larger and larger scales (the diffusion process)

* the notion of a cluster in the data set is quantified as a region in which the probability of escaping this region is low (within a certain time t)

* taking into account all linking between points, describing a global picture

]

---

# Diffusion Map

* dimensionality reduction/feature extraction algorithm

* coordinates can be computed from the eigenvectors and eigenvalues of a diffusion operator on the data(non-linear version of PCA)

* the Euclidean distance between points in the embedded space is equal to the "diffusion distance" between probability distributions centered at those points

* advantages:

  + robust to noise perturbation
  
  + emphasizing the concept of a cluster
  
  + computationally inexpensive

---

# Haghverdi 2016[2]: Model Design

* use Gaussians kernel to build the transition matrix

* use discrete random walks and graph searches to approximate the continuous process of diffusion

&lt;img src="./image/JC0522/paper1model.png" width="100%" style="display: block; margin: auto;" /&gt;

&lt;img src="./image/JC0522/paper1model2.png" width="90%" style="display: block; margin: auto;" /&gt;

---

# Haghverdi 2016[2]: Results

* Simulation study of robustness:

  + 100 bootstrap data sets
  
  + Kendall-tau correlation of each two bootstrap data sets

&lt;img src="./image/JC0522/paper1res.png" width="60%" style="display: block; margin: auto;" /&gt;

---

# Haghverdi 2016[2]: Results

* fig d: single-cell RNA (scRNA)-seq data of mouse  embryonic  stem  (ES)  cells  after leukemia inhibitory factor (LIF) withdrawal[3]

* fig f: an adult hematopoiesis scRNA-seq data set[4]

&lt;img src="./image/JC0522/paper1res2.png" width="80%" style="display: block; margin: auto;" /&gt;

---

# Farrell 2018[5]: Model Design

* use k-nearest-neighbor graph to build the transition matrix

* use discrete random walks and graph searches to approximate the continuous process of diffusion

&lt;img src="./image/JC0522/paper2model.png" width="100%" style="display: block; margin: auto;" /&gt;

*"the walks are biased to only transitionto cells of equal or earlier pseudotime so thatwhen they reach developmental branch points,they proceed toward the root and do not explore other cell types"*

---

# Farrell 2018[5]: Results

* 38,731 cells; 694 embryos; 

* 12 closely spaced stages of early zebrafish development(high blastula stage -&gt; six-somite stage)

&lt;img src="./image/JC0522/paper2res.png" width="100%" style="display: block; margin: auto;" /&gt;

---

# Farrell 2018[5]: Results

&lt;img src="./image/JC0522/paper2res2.png" width="80%" style="display: block; margin: auto;" /&gt;

???
recapitulated the developmental trajectories expected from classical embryological studies; found out some new branches

---

# Farrell 2018[5]: Results

&lt;img src="./image/JC0522/paper2res3.png" width="80%" style="display: block; margin: auto;" /&gt;

---

# Some Random Thoughts

* add pesudo-correlation within cells between different time point

* combine with RNA velocity -&gt; a new embedded space

---

class: center, middle

## Thanks!

---

# References



[1] R. R. Coifman, S. Lafon, A. B. Lee, et al. "Geometric diffusions as
a tool for harmonic analysis and structure definition of data:
Diffusion maps". In: _Proceedings of the national academy of sciences_
102.21 (2005), pp. 7426-7431.

[2] L. Haghverdi, M. Büttner, F. A. Wolf, et al. "Diffusion pseudotime
robustly reconstructs lineage branching". In: _Nature methods_ 13.10
(2016), p. 845.

[3] A. M. Klein, L. Mazutis, I. Akartuna, et al. "Droplet barcoding for
single-cell transcriptomics applied to embryonic stem cells". In:
_Cell_ 161.5 (2015), pp. 1187-1201.

[4] F. Paul, Y. Arkin, A. Giladi, et al. "Transcriptional heterogeneity
and lineage commitment in myeloid progenitors". In: _Cell_ 163.7
(2015), pp. 1663-1677.

[5] J. A. Farrell, Y. Wang, S. J. Riesenfeld, et al. "Single-cell
reconstruction of developmental trajectories during zebrafish
embryogenesis". In: _Science_ 360.6392 (2018).
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
