<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>DCATS: Differential Composition Analysis Transformed by a Similarity matrix in single-cell data</title>
    <meta charset="utf-8" />
    <meta name="author" content="Xinyi Lin" />
    <link href="libs/remark-css-0.0.1/default.css" rel="stylesheet" />
    <link href="libs/remark-css-0.0.1/metropolis.css" rel="stylesheet" />
    <link href="libs/remark-css-0.0.1/chocolate-fonts.css" rel="stylesheet" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# DCATS: Differential Composition Analysis Transformed by a Similarity matrix in single-cell data
### Xinyi Lin
### 2021/09/01 (updated: 2021-08-31)

---


# Workflow: previous version

&lt;img src="./image/LM0510/overall_pre.png" width="100%" style="display: block; margin: auto;" /&gt;

---

# Workflow

&lt;img src="./image/LM0510/overall.png" width="100%" style="display: block; margin: auto;" /&gt;

---

# Generalized Linear Models (GLMs)

Simple Linear Regression:

`$$E(Y) = \beta_0 + \beta\mathbf{X}$$`

where `\(y\)` follows normal distribution.

&lt;br/&gt;

Generalized Linear Models:

`$$g(\bar y) = \beta_0 + \beta\mathbf{X}$$`

where `\(y\)` follows other distribution including binomial distribution, poisson distribution.

---

# Base Model

* Type 1: Null model

Model 0: `\(g(\bar y) = \beta_0\)`

Model 1: `\(g(\bar y) = \beta_0 + \beta_1*tested\_covariate\)`

&lt;br/&gt;

* Type 2: Full model

Model 0: `\(g(\bar y) = \beta_0 + \beta*other\_covariates\)`

Model 1: `\(g(\bar y) = \beta_0 + \beta*other\_covariates + \beta_i*tested\_covariate\)`

---

# Model: Betabin GLM

A beta-binomial distirbution:

`$$P(Y=y|n,p) = \left(\begin{array}{c}n \\y \end{array} \right) = p^y(1-p)^{n-y}$$`
`$$f(p|a,b) = \frac{1}{B(a,b)}p^{a-1}(1-p)^{b-1}$$`

`$$E(Y|n,a,b) = E[E(Y|n,a,b,p)] = \frac{na}{a+b}$$`

`$$Var(Y|n,a,b) = E[Var(Y|n,a,b,p)] + Var[E(Y|n,a,b,p)] = \frac{nab(a+b+n)}{(a+b)^2(a+b+1)}$$`
Let `$$\pi = \frac{a}{a+b}, \phi = \frac{1}{a+b+1}$$`then `$$E(Y|n,\pi,\phi) = n\pi, Var(Y|n,\pi,\phi) = n\pi(1-\pi)[1+(n-1)\times\phi]$$`

---

# Over-dispersion term

* Without fixed over-dispersion term : 

   + `\(\phi\)` is estimated in each beta-binomial GLM for each cell type
   
&lt;br/&gt;

* With fixed over-dispersion term : 

   + `\(\phi\)` is estimated across all cell types before testing
   
   + is given in each beta-binomial GLM for each cell type

---

# Simulation Design

&lt;img src="./image/LM0510/dcats_simAll.png" width="90%" style="display: block; margin: auto;" /&gt;

---

# Theoretical Simulation

&lt;img src="./image/LM0901/figure1A.png" width="100%" style="display: block; margin: auto;" /&gt;

---

# Simulation with transcriptomics

&lt;img src="./image/LM0901/Fig1_B1.png" width="100%" style="display: block; margin: auto;" /&gt;

---

# Simulation with transcriptomics

&lt;img src="./image/LM0901/supFig4.png" width="100%" style="display: block; margin: auto;" /&gt;

---

# Simulation with covariates

* 10 samples for each condition 

* 8 cell types

* three covariates: condition, age, gender

   + condition: (6, 6, 10, 10, 8, 8, 8, 8) vs (8, 8, 8, 8, 8, 8, 8, 8)
   
   + age: (-0.05, 0, 0, 0.05, 0.05, 0, 0, -0.05) for one unit increase in age
   
   + gender: (-2, 2, 0, 0, -2, 2, 0, 0) for male versus female
   
   + additive effect
   
---

# Simulation with covariates

&lt;img src="./image/LM0901/Fig1_B2.png" width="100%" style="display: block; margin: auto;" /&gt;

---

# Real world data 1

&lt;img src="./image/LM0901/Fig2A.png" width="100%" style="display: block; margin: auto;" /&gt;

&lt;p style="font-family: times, serif; font-size:11pt; font-style:italic"&gt;
Haber, Adam L., et al. "A single-cell survey of the small intestinal epithelium." Nature 551.7680 (2017): 333-339.
&lt;/p&gt;

---

# Real world data 2

&lt;img src="./image/LM0901/Fig2B.png" width="50%" style="display: block; margin: auto;" /&gt;

&lt;p style="font-family: times, serif; font-size:11pt; font-style:italic"&gt;
Kang, Hyun Min, et al. "Multiplexed droplet single-cell RNA-sequencing using natural genetic variation." Nature biotechnology 36.1 (2018): 89-94.
&lt;/p&gt;
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
