<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>DCATS: Differential Composition Analysis Transformed by a Similarity matrix in single-cell data</title>
    <meta charset="utf-8" />
    <meta name="author" content="Xinyi Lin" />
    <link href="libs/remark-css-0.0.1/default.css" rel="stylesheet" />
    <link href="libs/remark-css-0.0.1/metropolis.css" rel="stylesheet" />
    <link href="libs/remark-css-0.0.1/chocolate-fonts.css" rel="stylesheet" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# DCATS: Differential Composition Analysis Transformed by a Similarity matrix in single-cell data
### Xinyi Lin
### 2021/05/08 (updated: 2021-05-17)

---


# Motivation

&lt;img src="./image/RPG_discussion/dcats_background.png" width="2800" height="10%" style="display: block; margin: auto;" /&gt;

---

# Workflow

&lt;img src="./image/LM0510/overall.png" width="100%" style="display: block; margin: auto;" /&gt;

Two main hypotheses:

* the clustering step is often inaccurate and may be systematically biased

* the reads counts follow beta-binomial distribution 

---

# Bias Correction

`\begin{bmatrix}
    m_{AA}       &amp; m_{AB} &amp; \dots &amp; m_{AK} \\
    m_{BA}       &amp; m_{BB} &amp; \dots &amp; m_{BK} \\
    \vdots &amp; \vdots &amp; \ddots &amp; \vdots \\
    m_{KA}       &amp; m_{KB} &amp; \dots &amp; m_{KK}
\end{bmatrix}`

`\(m_{ij}\)` denotes the probability of a cell `\(c\)` in type `\(i\)` be assigned to type `\(j\)` by the clustering method

Two procedures:

* Expectation–maximization(EM) algorithm

* Sampling based on the true proportions

---

# Expectation–maximization(EM) Algorithm

An approach for maximum likelihood estimation in the presence of latent variables

`$$l(\theta,\mathbf{Y}) = l(\theta, Y_{obs}, Z)$$`

* E-Step: Estimate the missing variables in the dataset.

`$$Q(\theta, \theta^{(t)}) = Ez[l(\theta, Y_{obs}, Z)|Y_{obs}, \theta^{(t)}]$$`

* M-Step: Maximize the parameters of the model in the presence of the data.

`$$\theta^{(t+1)} = arg\,max_{\theta}\,Q(\theta, \theta^{(t)})$$`

---

# Bias Correction: EM Part

`$$\begin{split}
		\mathcal{L(\mu)} &amp; = P(x | \mu, M)
		= \prod_{j=1}^{K} \prod_{c=1}^{x_j} \left[\sum_{i=1}^{K} P(I_c=i|\mu) P(A_c=j|I_c=i) \right]\\
		&amp; =  \prod_{j=1}^{K} \prod_{c=1}^{x_j} \left[ \sum_{i=1}^{K} \mu_i m_{i,j} \right] 
		= \prod_{j=1}^{K} \left[ \sum_{i=1}^{K} \mu_i m_{i,j} \right]^{x_j} 
	\end{split}$$`

`\(\mu\)` denotes unknown cell type composition vector, `\(x\)` denotes observing cell counts vector

Introducing a latent variable `\(z_{i,j}\)` denoting the expected probability of clustered cell type `\(i\)` coming from the real cell type `\(j\)`.

`$$z_{i,j} = P(I_c=j|A_c=i) = 
	\frac{P(A_c=i|I_c=j)P(I_c=j|\mu)}
	{\sum_{t=1}^{K} P(A_c=i|I_c=t)P(I_c=t|\mu)}
	= \frac{m_{j,i} \mu_j}{\sum_{t=1}^{K} m_{t,i}\mu_t}$$`

Through maximize `\(log\,\mathcal{L(z)}\)`, we can get `$$\mu_j = \frac{\sum_{i=1}^{K}z_{i,j} x_i}{\sum_{t=1}^{K}\sum_{i=1}^{K}z_{i,t} x_i}$$`

---

# Bias Correction: Sampling Process

`$$g_{i} \sim Multinomial(\mu, M)$$`

`\(g_{i}\)` denotes the `\(i\)`th sampled vector following multinomial distribution with size `\(\mu\)` and probability `\(M\)`

---

# Generalized Linear Models (GLMs)

Simple Linear Regression:

`$$E(Y) = \beta_0 + \beta\mathbf{X}$$`

* Random component: `\(Y\)`

* Systematic component: `\(\mathbf{X}\)`

* Link function: `\(E(Y_i)\)`

&lt;br/&gt;

Generalized Linear Models:

`$$g(\mu) = \beta_0 + \beta\mathbf{X}$$`

---

# Model: Betabin GLM

A beta-binomial distirbution:

`$$P(Y=y|n,p) = \left(\begin{array}{c}n \\y \end{array} \right) = p^y(1-p)^{n-y}$$`
`$$f(p|a,b) = \frac{1}{B(a,b)}p^{a-1}(1-p)^{b-1}$$`

`$$E(Y|n,a,b) = E[E(Y|n,a,b,p)] = \frac{na}{a+b}$$`

`$$Var(Y|n,a,b) = E[Var(Y|n,a,b,p)] + Var[E(Y|n,a,b,p)] = \frac{nab(a+b+n)}{(a+b)^2(a+b+1)}$$`
Let `$$\pi = \frac{a}{a+b}, \phi = \frac{1}{a+b+1}$$`then `$$E(Y|n,\pi,\phi) = n\pi, Var(Y|n,\pi,\phi) = n\pi(1-\pi)[1+(n-1)\times\phi]$$`

---

# Likelihood Ratio Test

Model with condition:

`$$g(\mu) = \beta_0+\beta_1*condition$$`

&lt;br/&gt;

`$$H_0: \beta_1 = 0, H_1:\beta_1 \neq 0$$`

`$$\lambda_{LR} = -2[l(\theta_0)-l(\hat\theta)]$$`
under the null hypothesis

`$$\lambda_{LR} \sim \chi^2$$`
---

# Confusion matrix

* Uniform type: 

`\begin{bmatrix}
    a       &amp; (1-a)/(K-1) &amp; \dots &amp; (1-a)/(K-1) \\
    (1-a)/(K-1)  &amp;   a &amp; \dots &amp; (1-a)/(K-1) \\
    \vdots &amp; \vdots &amp; \ddots &amp; \vdots \\
    (1-a)/(K-1)  &amp;   (1-a)/(K-1) &amp; \dots &amp; a
\end{bmatrix}`

* KNN type: `\(m_{ij}\)` = # of cluster `\(i\)`'s neighborhoods `\(\in\)` cluster `\(j\)`

* SVM type: 

&lt;img src="./image/LM0510/5foldCV.png" width="60%" style="display: block; margin: auto;" /&gt;

---

# Simulation Design

&lt;img src="./image/LM0510/dcats_simAll.png" width="90%" style="display: block; margin: auto;" /&gt;

---

# Theoretical Simulation

&lt;img src="./image/LM0510/res1.png" width="100%" style="display: block; margin: auto;" /&gt;

---

# scRNA-seq Simulation

&lt;img src="./image/RPG_discussion/dcats_result2.png" width="90%" style="display: block; margin: auto;" /&gt;

---

# Real World Data Analysis

Give consistent conclusions as original papers
  
&lt;img src="./image/LM0415/DCATS_fig1C.png" width="100%" style="display: block; margin: auto;" /&gt;

---

class: center, middle

## Thanks!
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
