<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>DCATS: Differential Composition Analysis of Single-Cell data</title>
    <meta charset="utf-8" />
    <meta name="author" content="Xinyi Lin" />
    <link href="libs/remark-css-0.0.1/default.css" rel="stylesheet" />
    <link href="libs/remark-css-0.0.1/nhsr.css" rel="stylesheet" />
    <link href="libs/remark-css-0.0.1/default-fonts.css" rel="stylesheet" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# DCATS: Differential Composition Analysis of Single-Cell data
### Xinyi Lin
### The University of Hong Kong, Ho Lab
### 2021/11/01

---


background-image: url("./image/hku2.jpg")
background-position: bottom left
background-size: 150px 30px
layout: true

&lt;style type="text/css"&gt;
.remark-slide-content {
    font-size: 17px;
    padding: 1em 4em 1em 4em;
}
&lt;/style&gt;

---

# Motivation

* R package designed for differential composition analysis on single cell data

* Basic assumptions

  + Cell counts follow beta-binomial distribution

  + Misclassification error exists[2]


&lt;img src="./image/BioC_Asia/simM.png" width="90%" style="display: block; margin: auto;" /&gt;
&lt;br/&gt;
&lt;small&gt;
cM, CD14+CD16– monocytes; ncM, CD14+CD16+ monocytes; DC, dendritic cells; Mkc, megakaryocytes; Th, CD4+ T cells; B, B cells; Tc, CD8+ T cells; NK, natural killer cells
&lt;small&gt;
---

# Workflow

&lt;br/&gt;

&lt;img src="./image/LM0510/overall.png" width="100%" style="display: block; margin: auto;" /&gt;

---

# Workflow

&lt;br/&gt;

&lt;img src="./image/BioC_Asia/overall_label.png" width="100%" style="display: block; margin: auto;" /&gt;

---

# Method: Misclassification Correction

* Uniform type: 

`\begin{bmatrix}
    a       &amp; (1-a)/(K-1) &amp; \dots &amp; (1-a)/(K-1) \\
    (1-a)/(K-1)  &amp;   a &amp; \dots &amp; (1-a)/(K-1) \\
    \vdots &amp; \vdots &amp; \ddots &amp; \vdots \\
    (1-a)/(K-1)  &amp;   (1-a)/(K-1) &amp; \dots &amp; a
\end{bmatrix}`

&lt;br/&gt;

* KNN type: `\(m_{ij}\)` = % of cluster `\(i\)`'s neighborhoods `\(\in\)` cluster `\(j\)`

&lt;br/&gt;

* SVM type: 

&lt;img src="./image/LM0510/5foldCV.png" width="90%" style="display: block; margin: auto;" /&gt;

---

# Method: Beta-binomial GLM

For each cell type:

.pull-left[

* Cell count vector

`$$\begin{align}
    y &amp;= \begin{bmatrix}
           y_{1} \\
           y_{2} \\
           \vdots \\
           y_{m}
         \end{bmatrix}
  \end{align}$$`
]

.pull-right[

* Design matrix

`$$\begin{bmatrix}
    x_{11} &amp; x_{12} &amp; \dots  &amp; x_{1n} \\
    x_{21} &amp; x_{22} &amp; \dots  &amp; x_{2n} \\
    \vdots &amp; \vdots &amp; \ddots &amp; \vdots \\
    x_{m1} &amp; x_{m2} &amp; \dots  &amp; x_{mn}
\end{bmatrix}$$`
]

&lt;br/&gt;

* With mean `\(\mu\)` of `\(y\)`, a beta-binomial GLM can be fitted as:  

`$$ln(\frac{\mu}{1-\mu}) = \beta_0 + \beta_1 \times \mathbf x_1 + \beta_2 \times \mathbf x_2 ......$$`
  
---

# Method: Models Selection

* Type 1: Null model

Model 0: `\(g(\mu) = \beta_0\)`

Model 1: `\(g(\mu) = \beta_0 + \beta_1*tested\_covariate\)`

&lt;br/&gt;

* Type 2: Full model

Model 0: `\(g(\mu) = \beta_0 + \beta*other\_covariates\)`

Model 1: `\(g(\mu) = \beta_0 + \beta*other\_covariates + \beta_i*tested\_covariate\)`

&lt;table&gt;
 &lt;thead&gt;
  &lt;tr&gt;
   &lt;th style="text-align:left;"&gt;   &lt;/th&gt;
   &lt;th style="text-align:right;"&gt; age &lt;/th&gt;
   &lt;th style="text-align:left;"&gt; sex &lt;/th&gt;
   &lt;th style="text-align:left;"&gt; sample_type &lt;/th&gt;
   &lt;th style="text-align:left;"&gt; condition &lt;/th&gt;
  &lt;/tr&gt;
 &lt;/thead&gt;
&lt;tbody&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; S-HC003 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 46 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; M &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; fresh PBMC &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; control &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; S-HC004 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 34 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; M &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; fresh PBMC &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; control &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; S-HC005 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 37 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; F &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; fresh PBMC &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; control &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; S-HC006 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 27 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; M &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; fresh PBMC &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; control &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; S-HC007 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 27 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; M &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; fresh PBMC &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; control &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; S-HC008 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 44 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; M &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; fresh PBMC &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; control &lt;/td&gt;
  &lt;/tr&gt;
&lt;/tbody&gt;
&lt;/table&gt;


---

# Method: Determine Over-dispersion

* Numbers of cells follow beta-binomial distribution:
   
`$$P(Y=y|n,p) = \left(\begin{array}{c}n \\y \end{array} \right) = p^y(1-p)^{n-y}$$`
`$$f(p|a,b) = \frac{1}{B(a,b)}p^{a-1}(1-p)^{b-1}$$`
`$$E(Y|n,\pi,\phi) = n\pi, Var(Y|n,\pi,\phi) = n\pi(1-\pi)[1+(n-1)\times\phi]$$`

&lt;br/&gt;

* Without fixed over-dispersion term : 

   + `\(\phi\)` is estimated in each beta-binomial GLM for each cell type
   
&lt;br/&gt;

* With fixed over-dispersion term : 

   + `\(\phi\)` is estimated across all cell types before testing
   
   + The estimated `\(\phi\)` is given in each beta-binomial GLM for each cell type


---

# How to Use DCATS

* **Count Matrix**


```r
data("Haber2017")
rbind(Haber2017$count_ctrl, Haber2017$count_Hpoly3)
```

```
##    Endocrine Enterocyte Enterocyte.Progenitor Goblet Stem  TA TA.Early Tuft
## B1        36         59                   136     36  239 125      191   18
## B2         5         46                    23     20   50  11       40    5
## B3        45         98                   188    124  250 155      365   33
## B4        26        221                   198     36  131 130      196    4
## B5        52         75                   347     66  323 263      313   51
## B6        65        126                   115     33   65  39      129   59
```

---

# How to Use DCATS

* **Design Matrix**


```r
sim_design = data.frame(condition = c(rep("control", 4), rep("Hpoly3", 2)))
print(sim_design)
```

```
##   condition
## 1   control
## 2   control
## 3   control
## 4   control
## 5    Hpoly3
## 6    Hpoly3
```

---

# How to Use DCATS

* **Design Matrix**


```r
data("Ren2021")
print(head(Ren2021$designM, 10))
```

```
##         age sex sample_type   state
## S-HC003  46   M  fresh PBMC control
## S-HC004  34   M  fresh PBMC control
## S-HC005  37   F  fresh PBMC control
## S-HC006  27   M  fresh PBMC control
## S-HC007  27   M  fresh PBMC control
## S-HC008  44   M  fresh PBMC control
## S-HC009  29   M  fresh PBMC control
## S-HC010  58   M  fresh PBMC control
## S-HC011  35   M  fresh PBMC control
## S-HC012  33   M  fresh PBMC control
```

---

# How to Use DCATS

* **Misclassification Matrix (a K `\(\times\)` K matrix)**

&lt;br/&gt;



&lt;img src="./image/BioC_Asia/svm_mat.png" width="90%" style="display: block; margin: auto;" /&gt;


---

# How to Use DCATS

* **Misclassification Matrix**


```r
data("Kang2017")
data("simulation")

# Three ways to calculate similarity matrices
## Uniform type
simil_mat = create_simMat(K = 3, confuse_rate = 0.2)

## KNN type
knn_mat = knn_simMat(KNN_matrix = simulation$knnGraphs, clusters = simulation$labels)

## SVM type
svm_mat = svm_simMat(dataframe = Kang2017$svmDF)
```

---

# How to Use DCATS

* **Main Function**



.pull-left[

```r
print(sim_count)
```

```
##      [,1] [,2] [,3]
## [1,]   36   35   29
## [2,]  271  279  250
## [3,]  518  379  403
## [4,]  152  220  228
## [5,]   84   87   79
## [6,]  259  203  238
## [7,]  345  376  379
```

]
.pull-right[

```r
print(sim_design)
```

```
##   condition gender
## 1        g1 Female
## 2        g1 Female
## 3        g1 Female
## 4        g1 Female
## 5        g2   Male
## 6        g2 Female
## 7        g2 Female
```
]


```r
## null model, flexible phi
res = dcats_GLM(sim_count, sim_design, similarity_mat = simil_mat)
## full model, flexible phi
res = dcats_GLM(sim_count, sim_design, simil_mat, base_model='FULL')
## null model, fixed phi
phi = getPhi(sim_count, sim_design)
res = dcats_GLM(sim_count, sim_design, simil_mat, fix_phi = phi)
```

---

# How to Use DCATS

* **Main Function**


```r
res = dcats_GLM(sim_count, sim_design, similarity_mat = simil_mat)
print(res$LRT_pvals)
```

```
##             condition    gender
## cell_type_1 0.8939193 0.9903591
## cell_type_2 0.6906093 0.6115926
## cell_type_3 0.5738177 0.6273852
```


---

# Method: Simulation Design

&lt;br/&gt;
&lt;br/&gt;

&lt;img src="./image/LM0510/dcats_simAll.png" width="100%" style="display: block; margin: auto;" /&gt;

---

# Results: Simulation

* **Theoretical Simulation**

&lt;br/&gt;

&lt;img src="./image/BioC_Asia/Fig1_A.png" width="100%" style="display: block; margin: auto;" /&gt;

---

# Results: Simulation

* **Simulation with Expression Info**

&lt;br/&gt;

&lt;img src="./image/BioC_Asia/Fig1_B.png" width="100%" style="display: block; margin: auto;" /&gt;

---

# Results: Simulation

* **Simulation with Expression Info**

&lt;img src="./image/BioC_Asia/supFig1B.png" width="100%" style="display: block; margin: auto;" /&gt;

---

# Results: Simulation

* **Simulation with Covariates**

&lt;img src="./image/BioC_Asia/Fig1_C.png" width="100%" style="display: block; margin: auto;" /&gt;

---

# Results: Real World Data[2]

&lt;br/&gt;
&lt;br/&gt;
&lt;br/&gt;

&lt;img src="./image/BioC_Asia/Fig2_A.png" width="100%" style="display: block; margin: auto;" /&gt;

&lt;br/&gt;

cM, CD14+CD16– monocytes; ncM, CD14+CD16+ monocytes; DC, dendritic cells; Mkc, megakaryocytes; Th, CD4+ T cells; B, B cells; Tc, CD8+ T cells; NK, natural killer cells

---

# Results: Real World Data[3]

&lt;br/&gt;

&lt;img src="./image/BioC_Asia/Fig2_BD.png" width="100%" style="display: block; margin: auto;" /&gt;

E, Enterocyte; TA, transit amplifying; TAE, TA.Early; EP, Enterocyte.Progenitor; Gob, Goblet

---

# Results: Real World Data[4]

&lt;br/&gt;

&lt;img src="./image/BioC_Asia/Fig2_E.png" width="90%" style="display: block; margin: auto;" /&gt;

---

# Acknowledgments

Joshua Ho Lab

Weizhong Zheng

Arron Kwok

Junyi Chen

All lab members
&lt;br/&gt;
&lt;br/&gt;
Yuanhua Huang Lab

All lab members

---

# References



[1] L. Zappia, B. Phipson, and A. Oshlack. "Splatter: simulation of
single-cell RNA sequencing data". In: _Genome biology_ 18.1 (2017), pp.
1-15.

[2] H. M. Kang, M. Subramaniam, S. Targ, et al. "Multiplexed droplet
single-cell RNA-sequencing using natural genetic variation". In:
_Nature biotechnology_ 36.1 (2018), p. 89.

[3] A. L. Haber, M. Biton, N. Rogel, et al. "A single-cell survey of
the small intestinal epithelium". In: _Nature_ 551.7680 (2017), pp.
333-339.

[4] X. Ren, W. Wen, X. Fan, et al. "COVID-19 immune features revealed
by a large-scale single-cell transcriptome atlas". In: _Cell_ 184.7
(2021), pp. 1895-1913.

---

# Results: Simulation

* **Theoretical Simulation**

&lt;img src="./image/BioC_Asia/supFig1A.png" width="100%" style="display: block; margin: auto;" /&gt;
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
