<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>First Year RPG Discussion</title>
    <meta charset="utf-8" />
    <meta name="author" content="Xinyi Lin" />
    <link href="libs/remark-css-0.0.1/default.css" rel="stylesheet" />
    <link href="libs/remark-css-0.0.1/metropolis.css" rel="stylesheet" />
    <link href="libs/remark-css-0.0.1/chocolate-fonts.css" rel="stylesheet" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# First Year RPG Discussion
### Xinyi Lin
### Supervisor: Joshua Ho
### 2021/03/31 (updated: 2021-04-12)

---


# Background &amp; Research Goals

* Challenges on single cell data analysis

  + lack of accurate cell clustering method
  
  + hard to observe dynamic changes on different cell types

  + inaccurate trajectory analysis

  + needs of computational tools for different data types analysis
  
* PhD focus

  statistical analysis for scalable single cell analysis
  
  + statistical modeling
  
  + single cell data analysis

---

# DCATS: Background

* Tissues are cellular heterogeneity

* Composition difference exists

* Interested in changes of proportions in dynamic process

&lt;img src="./image/RPG_discussion/dcats_background.png" width="2800" height="10%" style="display: block; margin: auto;" /&gt;

---

# DCATS: Design

* R package designed for differential composition analysis on single cell data

* Basic assumptions

  + misclassification error exists

  + cell counts follow binomial distribution
  
* Input

  + cell counts matrices of samples coming from different conditions
  
  + a confusion matrix describing the misclassification error(user-input or calculated by DCATS)

---

# DCATS: Theoretical Simulation

* Given a similarity matrix, only simulate cell count matrices

* For each replicates:

`$$(p_1, p_2, p_3) \sim Dir(\alpha_1, \alpha_2, \alpha_3)$$`

`$$(c_1, c_2, c_3) \sim multinomial(p_1,p_2,p_3|M_{sim})$$`
* Result

&lt;img src="./image/RPG_discussion/dcats_result1.png" width="80%" style="display: block; margin: auto;" /&gt;

---

# DCATS: Simulation with Gene Expression Information

* Simulation process 

&lt;img src="./image/RPG_discussion/dcats_sim.png" width="90%" style="display: block; margin: auto;" /&gt;

---

# DCATS: Simulation with Gene Expression Information

* Simulation results

&lt;img src="./image/RPG_discussion/dcats_result2.png" width="90%" style="display: block; margin: auto;" /&gt;

---

# DCATS: Simulation with Gene Expression Information

* Real world data analysis

  + Give consistent conclusions as original papers:     *Kang, Hyun Min, et al, 2018* and *Haber, Adam L., et al, 2017*

---

# Current Projects: other side projects(will be updated)

* Giant Cell Tumor of Bones(GCTB)

  + a somatic alterations(*H3F3A*, Gly34Val): IGV, cellSNP
  
  + trajectory analysis
  
&lt;div class="figure" style="text-align: center"&gt;
&lt;img src="./image/RPG_discussion/GCTB_res1.png" alt="Trajectory of myeloid cells" width="465" /&gt;
&lt;p class="caption"&gt;Trajectory of myeloid cells&lt;/p&gt;
&lt;/div&gt;

---

# Current Projects: other side projects(will be updated)

* Mouse's Kidneys

  + give a possible clustering result 
  
  + identify some cell types based on the clustering result
  
&lt;div class="figure" style="text-align: center"&gt;
&lt;img src="./image/RPG_discussion/kidney_res1.jpg" alt="Clustering result with notation" width="55%" /&gt;
&lt;p class="caption"&gt;Clustering result with notation&lt;/p&gt;
&lt;/div&gt;

---

# Future Plan: current projects

* DCATS: get it published

* Side projects

  + be familiar with computational tools related to scRNA-seq
  
  + finish related bioinformatics analysis
  
  + find out some new problems during collaboration
  
  + learn more biological knowledge

---

# Future Plan: exchange program

* Time: whenever travel restriction eases

* Duration: 3-6 months

* Lab: [Xi Lab](https://sph.pku.edu.cn/info/1646/4192.htm), Peking University

* Goals: Collaborate with people there on some scRNA-seq related projects

---

# Future Plan: Other possible projects

* cross-species analysis

* time-series analysis

* other possible statistical modeling

---

# Acknowledge

* Dr. Joshua Ho

* Other labmates on [Ho Lab, HKU](https://holab-hku.github.io/)

* Dr. Yuanhua Huang and other people in [Huang Lab, HKU](https://web.hku.hk/~yuanhua/)

* Shan Han from  Dr. Kelvin Cheung's lab

* Professor Sydney Tang and postdoc Stella Yiu

---

# References



[1] L. Zappia, B. Phipson, and A. Oshlack. "Splatter: simulation of
single-cell RNA sequencing data". In: _Genome biology_ 18.1 (2017), pp.
1-15.

[2] H. M. Kang, M. Subramaniam, S. Targ, et al. "Multiplexed droplet
single-cell RNA-sequencing using natural genetic variation". In:
_Nature biotechnology_ 36.1 (2018), p. 89.

[3] A. L. Haber, M. Biton, N. Rogel, et al. "A single-cell survey of
the small intestinal epithelium". In: _Nature_ 551.7680 (2017), pp.
333-339.

---

class: center, middle

## Thanks!
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
